Lib/ast.py

```diff
@@ -1199,8 +1199,11 @@ def _write_docstring(self, node):
 
     def _write_constant(self, value):
         if isinstance(value, (float, complex)):
-            # Substitute overflowing decimal literal for AST infinities.
-            self.write(repr(value).replace("inf", _INFSTR))
+            # Substitute overflowing decimal literal for AST infinities,
+            # and inf - inf for NaNs.
+            self.write(repr(value)
+                .replace("inf", _INFSTR)
+                .replace("nan", "({}-{})".format(_INFSTR, _INFSTR)))
```

Comment:
Nit: this would read better and be slightly faster as an f-string.

Reply:
Agreed, perhaps even as a constant (`_NANSTR` so that we won't create it every time)

Reply:
> Nit: this would read better and be slightly faster as an f-string.

Fair; I tend to forget that f-strings exist.